nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/leader/election/CuratorLeaderElectionManager.java

```diff
@@ -446,27 +446,51 @@ public String getParticipantId() {
         private final String participantId;
 
         private volatile boolean leader;
+        private long leaderUpdateTimestamp = 0L;
+        private final long MAX_CACHE_MILLIS = TimeUnit.SECONDS.toMillis(60L);
```

Comment:
@markap14 overall this looks like a good change/idea.  How about though we only increase the time at which it could be wrong on its role to no more than 5 seconds for now?  We can learn from its behavior over time and decide whether to increase or decrease?

Reply:
Sure, that's fine. Will update that.